### ISSUE TITLE
Logistic output layer training slowly

### ISSUE THREAD

**Issue by weixsong** [openai/pixel-cnn#28]
Hi, 

As the paper mentioned, the Logistic output layer aim to reduce the computation complexity of Softmax output layer.

But I found that using the Logistic output layer, the model convergence slowly for my network, not piexlcnn, a different network.

The logistic output layer works for my network, but it really take a more longer time for the model to convergence.

Is there any suggestion for the convergence of Logistic output layer?

The logistic quantization is more than 256, say 2^12 or 2^16.

**Comment by bliep** [comment]
The minimum 'scale' size should reflect the quant step, so min scale should be in the order 1/nr_quants or it won't converge beyond the 8 bit resolution (for 16 bits change the minimum of log_scale from -7 to -14). Oh and btw, I get smoother training loss when training on the scale itself, instead of the log_scale.

**Comment by weixsong** [comment]
@bliep , thanks very much.
